tests/rptest/tests/partition_movement_test.py

```diff
@@ -990,21 +992,44 @@ def _finish_upgrade(self, num_upgraded_already: int):
     @skip_fips_mode
     @skip_debug_mode  # rolling restarts require more reliable recovery that a slow debug mode cluster can provide
     @cluster(num_nodes=5, log_allow_list=PREV_VERSION_LOG_ALLOW_LIST)
-    @matrix(num_to_upgrade=[0, 2], cloud_storage_type=get_cloud_storage_type())
-    def test_shadow_indexing(self, num_to_upgrade, cloud_storage_type):
+    @matrix(
+        num_to_upgrade=[0, 2],
+        cloud_storage_type=get_cloud_storage_type(),
+        with_cloud_topics=[True, False],
+    )
+    def test_shadow_indexing(
+        self, num_to_upgrade, cloud_storage_type, with_cloud_topics: bool
+    ):
         """
         Test interaction between the shadow indexing and the partition movement.
         Partition movement generate partitions with different revision-ids and the
         archival/shadow-indexing subsystem is using revision to generate unique object
         keys inside the remote storage.
         """
-        throughput, records, moves, partitions = self._get_scale_params()
+        test_mixed_versions: bool = num_to_upgrade > 0
+        if test_mixed_versions and with_cloud_topics:
+            # Upgrades not supported with cloud topics yet.
+            # Avoid the "Test requested 5 nodes, used only 0" error.
+            self.logger.warn("skipping test, upgrades with cloud topics not supported")
+            self.redpanda = make_redpanda_service(self.test_context, 0)
+            self.test_context.cluster.alloc(ClusterSpec.simple_linux(5))
+            return
+        extra_rp_conf: dict = {}
+        if with_cloud_topics:
+            extra_rp_conf = {"cloud_topics_enabled": True}
 
-        test_mixed_versions = num_to_upgrade > 0
+        throughput, records, moves, partitions = self._get_scale_params()
         install_opts = InstallOptions(install_previous_version=test_mixed_versions)
-        self.start_redpanda(num_nodes=3, install_opts=install_opts)
+        self.start_redpanda(
+            num_nodes=3, install_opts=install_opts, extra_rp_conf=extra_rp_conf
+        )
 
-        spec = TopicSpec(name="topic", partition_count=partitions, replication_factor=3)
+        spec = TopicSpec(
+            name="topic",
+            partition_count=partitions,
+            replication_factor=3,
+            cloud_topics_enabled=with_cloud_topics,
+        )
         self.client().create_topic(spec)
```

Comment:
i dont' think this will enable cloud topics. even though the TopicSpec lets you set that flag, the topic creation interfaces that take the spec don't necessarily set that option. In particular, that client() maps to the Kafka Apache CLI client and it can't even be used to set cloud topics--you'll have to switch to the RPK client.

It might be useful to add a little safety guard so that other people don't think they are enabling things that aren't being enabled :/